Model/Service/Sync/AsyncBulkConsumer.php

```
@@ -0,0 +1,140 @@
+use Nosto\Tagging\Logger\Logger;
+use Nosto\Tagging\Model\Product\Index\IndexRepository;
+use Nosto\Tagging\Model\Service\Sync\SyncService as NostoSyncService;
+use Nosto\Tagging\Helper\Scope as NostoScopeHelper;
+
+/**
+ * Asynchronous Bulk Consumer
+ *
+ * Class AsyncBulkConsumer
+ */
+class AsyncBulkConsumer
+{
+    /** @var Logger */
+    private $logger;
+
+    /** @var JsonHelper */
+    private $jsonHelper;
+
+    /** @var IndexRepository */
+    private $indexRepository;
+
+    /** @var NostoSyncService */
+    private $nostoSyncService;
+
+    /** @var NostoScopeHelper */
+    private $nostoScopeHelper;
+
+    /** @var EntityManager */
+    private $entityManager;
+
+    /**
+     * AsyncBulkConsumer constructor.
+     *
+     * @param Logger $logger
+     * @param JsonHelper $jsonHelper
+     * @param IndexRepository $indexRepository
+     * @param NostoSyncService $nostoSyncService
+     * @param NostoScopeHelper $nostoScopeHelper
+     * @param EntityManager $entityManager
+     */
+    public function __construct(
+        Logger $logger,
+        JsonHelper $jsonHelper,
+        IndexRepository $indexRepository,
+        NostoSyncService $nostoSyncService,
+        NostoScopeHelper $nostoScopeHelper,
+        EntityManager $entityManager
+    ) {
+        $this->logger = $logger;
+        $this->jsonHelper = $jsonHelper;
+        $this->indexRepository = $indexRepository;
+        $this->nostoSyncService = $nostoSyncService;
+        $this->nostoScopeHelper = $nostoScopeHelper;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * Processing operation for product sync
+     *
+     * @param array|\Magento\AsynchronousOperations\Api\Data\OperationInterface $operation
+     * @return void
+     * @throws Exception
+     * @suppress PhanUndeclaredClassConstant
+     */
+    public function processOperation($operation)
+    {
+        $errorCode = null;
+        $message = null;
+        if (!is_array($operation)) {
```

Should we first check if `$operation` is an array. If not check that the type is `OperationInterface` and if that fails we should throw an exception.